dali/aux/optical_flow/optical_flow_adapter.h

```diff
@@ -0,0 +1,54 @@
+// Copyright (c) 2019, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef DALI_AUX_OPTICAL_FLOW_OPTICAL_FLOW_ADAPTER_H_
+#define DALI_AUX_OPTICAL_FLOW_OPTICAL_FLOW_ADAPTER_H_
+
+#include <dali/pipeline/data/backend.h>
+#include "dali/kernels/tensor_view.h"
+
+namespace dali {
+namespace optical_flow {
+
+enum VectorGridSize {
+  UNDEF,
+  SIZE_4,  /// 4x4 grid
+  MAX,
+};
+
+struct OpticalFlowParams {
+  float perf_quality_factor = .0f;  /// 0..1, where 0 is best quality, lowest performance
+  VectorGridSize grid_size = UNDEF;
+  bool enable_hints = false;
+};
+
+
+class DLL_PUBLIC OpticalFlowAdapter {
+ public:
+  explicit OpticalFlowAdapter(OpticalFlowParams params) {}
+
+
+  virtual void CalcOpticalFlow(kernels::TensorView<GPUBackend, const uint8_t, 3> reference_image,
+                               kernels::TensorView<GPUBackend, const uint8_t, 3> input_image,
+                               kernels::TensorView<GPUBackend, float, 1> output_image,
```

Comment:
```suggestion
                               kernels::TensorView<GPUBackend, FLOW_VECTOR, 2> output_image,
```
where `FLOW_VECTOR` is probably a template param, 
```
template <typename T>
struct FLOW_VECTOR {
   T flow_x;
   T flow_y;
};
```
We can probably default to floats like in OpenCV case, so we have a unified interface.

Certainly this should not be one-dimensional.

Reply:
Other option would be to just have a 
```suggestion
                               kernels::TensorView<GPUBackend, float, 3> output_image,
```
And the last dim would have a `size = 2`.